custom/git-app-deployer/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/litmuschaos/test-tools/pkg/log"
+	"github.com/openebs/maya/pkg/util/retry"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+
+	var kubeconfig *string
+	var timeout int
+	file := flag.String("file", "", "absolute path of the sock-shop")
+	flag.Parse()
+
+	if *file == "" {
+		*file = "week"
+		timeout = 180
+	} else {
+		timeout = 400
+	}
+	config, err := rest.InClusterConfig() // If In-Cluster is nil then it will go for Out-Cluster config
+	if config == nil {
+		//To get Out-Cluster config
+		if home := homedir.HomeDir(); home != "" {
+			kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "kubeconfig file it is out-of-cluster")
+		} else {
+			kubeconfig = flag.String("kubeconfig", "", "Path to the kubeconfig file")
+		}
+		//panic(err.Error())
+		flag.Parse()
+
+		// uses the current context in kubeconfig
+		config, err = clientcmd.BuildConfigFromFlags("", *kubeconfig)
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+	// use the current context in kubeconfig
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	log.Info("[Status]: App-deployer is started...")
+	log.Info(`[Status]: App-deployer type is: ` + *file + `-sock-shop.`)
+	if err := CreateNamespace("sock-shop"); err != nil {
+		log.Fatalf("Namespace sock-shop is already exist: %v", err)
+	}
+	log.Info("[Status]: Namespace shock-shop is successfully created!")
+
+	// /var/run/
+	if err := CreateShockShop(`/var/run/` + *file + `-sock-shop.yaml`); err != nil {
+		log.Fatalf("Error with creating experiment: %v", err)
+	}
+	log.Info("[Status]: sock-shop experiment is successfully created!")
+
+	if err := CheckApplicationStatus("sock-shop", "app=sock-shop", timeout, 2, clientset); err != nil {
+		log.Fatalf("err: %v", err)
+	}
+
+}
+
+//CreateNamespace ...
+func CreateNamespace(ns string) error {
+	//kubectl create ns sock-shop
+	command := exec.Command("kubectl", "create", "ns", "sock-shop")
```

Comment:
we are taking ns as an argument. can we use that value instead of hard-coding it?

Reply:
Thank you @ispeakc0de and @uditgaurav. I have updated it